osu.Game/Skinning/RulesetSkinProvidingContainer.cs

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System.Linq;
+using osu.Framework.Allocation;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Game.Beatmaps;
+using osu.Game.Rulesets;
+
+namespace osu.Game.Skinning
+{
+    /// <summary>
+    /// A type of <see cref="SkinProvidingContainer"/> that provides access to the beatmap skin and user skin,
+    /// each transformed with the ruleset's own skin transformer individually.
+    /// </summary>
+    public class RulesetSkinProvidingContainer : SkinProvidingContainer
```

Comment:
Weird having only `Ruleset` in the name when it's also handling the beatmap.

Also is it feasible to just pass in the `WorkingBeatmap` rather than the two separate components of it?

Reply:
Not sure, two separate `WorkingBeatmap`s are used in usages so I wasn't sure about doing that.

As for the name, yeah, I'm not sure, I chose `Ruleset` because it is the most benefited area with this container, maybe `Gameplay`? not sure if it's fine to have a "`Gameplay`" component in editor though

Reply:
Ruleset is better than Gameplay. Let's just go with it for now.